cg.lua

```diff
@@ -42,6 +42,11 @@ function optim.cg(opfunc, x, config, state)
    local ratio = config.ratio or 100
    local maxEval = config.maxEval or maxIter*1.25
    local red = 1
+   local cuda = torch.typename(x) == 'torch.CudaTensor'
```

Comment:
instead of this, the very simple change you have to do is:

local df1 = state.df1 or x.new()